### ISSUE TITLE
Security audit and internal training

### ISSUE THREAD

**Issue by prioux** [aces/cbrain#480]
All devs should read this document, then we should do a new security audit of the entire code base. The last one was too long ago.

http://guides.rubyonrails.org/security.html

**Comment by crocodoyle** [comment]
The second link is not working, do you have a cached version of the article you could post?

**Comment by prioux** [comment]
Yeah, looks like the site for phrack has been blocked or is down. It's good that I have a copy of the article. It's plain text. I'll not post it here, ask me tomorrow.

**Comment by shots47s** [comment]
So, I am assuming this wasn't done... should we schedule this before or after the Rails 5.0 upgrade?

**Comment by crocodoyle** [comment]
We did the training and fixed a couple of edge cases where cross-site scripting could be a problem but a full audit was not done

**Comment by prioux** [comment]
After. Any changes we make to the 4.* branch of CBRAIN means retrofitting thoses changes into the R5 code base.

**Comment by shots47s** [comment]
We should schedule, between @natacha-beck and @prioux 

Only need to review the **controllers**

**Comment by prioux** [comment]
I'm reading all this now, then I'll provide a summary to all developers.

**Comment by prioux** [comment]
The phrack article is back online by the way.
